### ISSUE TITLE
[FEATURE] FakeLag major improvement

### ISSUE THREAD

**Issue by be4dev** [CCBlueX/LiquidBounce#2509]
### LiquidBounce Branch

Nextgen

### Describe your feature request.

fakelag stops sending packets for some time, so we can actually change them based on the enemy actions, to achive combat advantage
1. we stopped sending packets to the server
2. enemy gets closer
3. client finds out that in the past 500ms we have desynced for we could have gotten to the block (x, y, z), which enemy isn't looking at (and so he won't be able to hit us for some time)
4. client recreates all those packets (we are still desynced)
5. client sends those packets and sets our position according to them
6. we get to the block (x, y, z). Enemy isn't looking there. We get 3-5 additional hits

so basically it finds the best move order and applies them

another good example is spleef
1. we are standing at block (x1, y1, z1), fakelag starts it's work
2. enemy gets close enough and breaks a few blocks (x1, y1, z1), (x2, y1, z1)
3. client finds out which way it should have moved in order not to fall for example to block (x1, y1, z2)
4. client recreates those steps and send those movement packets
5. server thinks that we were lagging and we actually performed moves to save ourself
6. we continue our gaming

there are tons of ways to use this (avoid arrows, combat advantage, tnt avoiding and much more)
